component/main.jsonnet

```diff
@@ -5,6 +5,49 @@ local inv = kap.inventory();
 // The hiera parameters for the component
 local params = inv.parameters.metallb;
 
+local namespace = kube.Namespace(params.namespace) {
+  metadata+: {
+    labels: {
+      app: 'metallb',
+    }
+  },
+};
+
+local memberlist_secret = kube.Secret(params.speaker.secretname) {
+  metadata+: {
+    namespace: params.namespace,
+  },
+  // need to use stringData here for secret reveal to work
+  stringData+: {
+    secretkey: params.memberlist_secretkey,
+  },
+};
+
+// Services
+local ingress_nginx_service = kube.Service('ingress-nginx') {
```

Comment:
Shouldn't this be handled by another component / Rancher itself?

Reply:
I agree, this does not really fit in this component.

Reply:
Removed.